### PR TITLE
Closes #536: Changed product addition layout

### LINKED ISSUE #536: Make the product addition form look like an empty product
- It should look like an empty product
- It should try to send pictures as soon as they are taken (or later if offline mode is enabled)
- It should let people add product name and brand
- It should not have an explicit save button : people wanting to add text should add it by editing and saving.

### PR DESCRIPTION
**Pull request by jaztriumph**
## Description

Changed the product addition layout to empty product. Made a new fragment for summary and made changes in ingredients and nutrition fragments so that they can be used for product addition. Now the images upload as soon as the pic is taken or when network is connected. Until now cropped images were were stored as temporary files in cache, I made changes to store them in pictures in apps directory, so that they will be accessible all time in offline edit.   

## Related issues and discussion
closes #536 
 
 ## Screen-shots, if any
 
pdt_addition
pdt_addition2
pdt-addition

### CODE REVIEW

**Comment by jaztriumph** [comment]
@Karljoones @policorn please review this PR.

**Comment by Karljoones** [comment]
@jaztriumph before we can have a look at this, can you check out the conflicts?

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -182,7 +182,9 @@
     <string name="txtTakePicture">Take picture</string>
     <string name="txtSendAll">Upload all</string>
     <string name="txtNoData">No data</string>
-    <string name="txtPictureNeeded">Please take a picture first</string>
+    <string name="txtPictureNeededDialogContent">Please take a picture to save changes</string>
```
Comment (Karljoones) [inline app/src/main/res/values/strings.xml]: Can you change this to something along the lines of
> You should add atleast one picture before submitting this product

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -182,7 +182,9 @@
     <string name="txtTakePicture">Take picture</string>
     <string name="txtSendAll">Upload all</string>
     <string name="txtNoData">No data</string>
-    <string name="txtPictureNeeded">Please take a picture first</string>
```
Comment (Karljoones) [inline app/src/main/res/values/strings.xml]: Can you change this to something along the lines of:
> Wait... 

Please use &#8230; for the ellipsis

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java [hunk 3]
```diff
@@ -263,7 +283,14 @@ public void onFailure(Call<State> call, Throwable t) {
                         .negativeText(R.string.txtNo)
                         .onPositive((dialog, which) -> {
                                     Intent intent = new Intent(activity, SaveProductOfflineActivity.class);
-                                    intent.putExtra("barcode", barcode);
+//                                    intent.putExtra("barcode", barcode);
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java]: Clean up please

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java [hunk 4]
```diff
@@ -197,7 +210,14 @@ public void onResponse(Call<State> call, Response<State> response) {
                             .negativeText(R.string.txtNo)
                             .onPositive((dialog, which) -> {
                                 Intent intent = new Intent(activity, SaveProductOfflineActivity.class);
-                                intent.putExtra("barcode", barcode);
+//                                intent.putExtra("barcode", barcode);
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java]: Clean up please

**Comment by jaztriumph** [comment]
@Karljoones  made the changes.

**Comment by Karljoones** [comment]
Thanks @jaztriumph for making most of the changes, however, there are a couple there in regards to strings that were not changed, could you have a look at these too?

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (jaztriumph) [reply app/src/main/res/values/strings.xml]: @Karljoones  should I assign  "Wait..." to `txtPictureNeeded`. Is it what you meant ?

**Comment by jaztriumph** [comment]
@Karljoones made changes.

**Comment by jaztriumph** [comment]
@Karljoones can you review this once again? If not I will have to resolve merge conflicts again :sweat_smile:

**Comment by jaztriumph** [comment]
@Karljoones there is no progress on this PR for past 3 days.

**Comment by jaztriumph** [comment]
@Karljoones @policorn @itchix can you review this once again :sweat_smile:

**Comment by jaztriumph** [comment]
@Karljoones  resolved merge conflicts.

**Comment by jaztriumph** [comment]
@karljoones please check this PR.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (Karljoones) [reply app/src/main/res/values/strings.xml]: Yes that is correct

Comment (jaztriumph) [reply app/src/main/res/values/strings.xml]: @Karljoones then all changes are made. I think this PR needs to be approved.

**Comment by jaztriumph** [comment]
@policorn @ross-holloway94 reviews.

**Comment by Karljoones** [comment]
@jaztriumph @teolemon let's get the conflicts resolved and merge this

**Comment by ross-holloway94** [comment]
@jaztriumph Try merging master into this branch again, and it should build successfully.

**Comment by jaztriumph** [comment]
@Karljoones Travis has finally passed.

**Pull request merged**